### PR TITLE
[walkers] ensure one cache per walker class

### LINKED ISSUE #266: Unusual use of `__new__()` in `NodeWalker`
Hunting for old-style `super()` calls I bumped into the `__new__()` method of the `NodeWalker` class:
```python
class NodeWalker:
    def __new__(cls, *args, **kwargs):
        cls._walker_cache = {}
        return super(NodeWalker, cls).__new__(cls)
```
As the class does not inherit from other classes the use of `super()` seems redundant here as it is equivalent to simply calling `cls()` but this is not what is unusual. What is unusual is modifying the class definition in the `__new__()` method. This means that all `NodeWalker` instances share the same `_walker_cache` dictionary and that this dictionary in reset every time a new `NodeWalker` instance is created. I cannot find a reason why this could be desirable.

Shouldn't this `__new__()` method be turned into a `__init__()` method?

### PR DESCRIPTION
**Pull request by apalala**
fixes #266

### CODE REVIEW

File: tatsu/walkers.py [hunk 1]
```diff
@@ -8,10 +8,18 @@
...
+        class_ = super().__new__(cls, name, bases, dct)
+        class_._walker_cache = {}
+        return class_
+
+
+class NodeWalker(metaclass=NodeWalkerMeta):
+    def __init__(self):
+        super().__init__()
+        # copy the class attribute to avoid linter warnings
+        self._walker_cache = type(self)._walker_cache  # pylint: disable=no-member
```
Comment (dnicolodi) [inline tatsu/walkers.py:22]: How is this any different than the simpler `self._walker_cache = {}` ?

Comment (dnicolodi) [reply tatsu/walkers.py:22]: Please disregard the previous comment, I was confused by the comment in the code. The cache dictionary is not copied.

Comment (apalala) [reply tatsu/walkers.py:22]: Please elaborate? Which do you think should be the text of the comment?

Comment (dnicolodi) [reply tatsu/walkers.py:22]: Reading the comment in the code, I though that what the code did was something like this:
```python
self._walker_cache = type(self)._walker_cache.copy()
```
I don't think the comment is wrong, it is just that the code pattern is a very unusual and I would prefer if there would be another way to silence the pylint warnings.

Comment (apalala) [reply tatsu/walkers.py:22]: I think that the patch doesn't hurt, it is inexpensive, and it silences all the `pylint` warnings without changing the existing code.

**Pull request merged**